models/pancakeswap/bnb/pancakeswap_v3_bnb_amm_trades.sql

```diff
@@ -52,22 +52,22 @@ SELECT
         when lower(erc20a.symbol) > lower(erc20b.symbol) then concat(erc20b.symbol, '-', erc20a.symbol)
         else concat(erc20a.symbol, '-', erc20b.symbol)
     end as token_pair
-    ,dexs.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount
-    ,dexs.token_sold_amount_raw / power(10, erc20b.decimals) AS token_sold_amount
-    ,CAST(dexs.token_bought_amount_raw AS DECIMAL(38,0)) AS token_bought_amount_raw
-    ,CAST(dexs.token_sold_amount_raw AS DECIMAL(38,0)) AS token_sold_amount_raw
+    ,CAST(dexs.token_bought_amount_raw AS UINT256) / power(10, erc20a.decimals) AS token_bought_amount
```

Comment:
any reason we explicitly cast to `unit256` in all these lines in this model, while the `v2 amm` script above doesn't? is it due to source decoded tables having differing datatypes between v2/v3?

Reply:
Looks like `amount` columns in v2 swap event tables are already `unit256 ` while in v3 `amount` columns are `int256`
<img width="357" alt="Screenshot 2023-07-11 at 11 23 05 AM" src="https://github.com/duneanalytics/spellbook/assets/129702169/60e62cfd-ed7e-4639-8f0f-b5f23d4c1184">
<img width="359" alt="Screenshot 2023-07-11 at 11 22 43 AM" src="https://github.com/duneanalytics/spellbook/assets/129702169/58cf9bc7-4d57-49c8-a24c-c934cfaad94b">
<img width="357" alt="Screenshot 2023-07-11 at 11 20 41 AM" src="https://github.com/duneanalytics/spellbook/assets/129702169/ae4f6fa5-96c4-4058-93fb-78decc4dec20">
<img width="376" alt="Screenshot 2023-07-11 at 11 20 17 AM" src="https://github.com/duneanalytics/spellbook/assets/129702169/8fb9ded7-d363-4229-ad3c-646cd8b7d62e">